modules/packages/Crypto.chpl

```diff
@@ -0,0 +1,953 @@
+/*
+ * Copyright 2004-2017 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*  A cryptographic library based on OpenSSl, targeted at flexible encryption
+    purposes.
+
+    The Crypto module focuses on providing various cryptographic utilities such as
+
+    - Symmetric Ciphers
+    - Asymmetric Ciphers
+    - Hashing Functions
+    - Key Derivation Functions
+    - Cryptographically-Secure Pseudo-Random Number Generators
+    - Miscellaneous Utilities such as CryptoBuffer, Envelopes, etc.
+
+    .. note::
+
+      This module is currently under development and will expand significantly
+      in upcoming releases.
+
+*/
+
+module Crypto {
+
+  use C_OpenSSL;
+
+  pragma "no doc"
+  proc generateKeys(bits: int) {
+   var localKeyPair: EVP_PKEY_PTR;
+   var keyCtx = EVP_PKEY_CTX_new_id(EVP_PKEY_RSA: c_int,
+                                    c_nil: ENGINE_PTR);
+   EVP_PKEY_keygen_init(keyCtx);
+   EVP_PKEY_CTX_set_rsa_keygen_bits(keyCtx, bits: c_int);
+   EVP_PKEY_keygen(keyCtx, localKeyPair);
+
+   EVP_PKEY_CTX_free(keyCtx);
+   return localKeyPair;
+  }
+
+  class CryptoBuffer {
```

Comment:
It would be helpful to include a doc-string for each class, e.g. 

```chpl
/* The CryptoBuffer is .... 

*/
class CryptoBuffer {
...
}

Reply:
@saru95 - are you intending to add these type of comments, or did you decide against this for this PR?

Reply:
I will in this PR itself. I want to resolve the errors first.